zipkin/src/main/java/com/linecorp/armeria/server/tracing/HttpTracingService.java

```diff
@@ -84,10 +85,16 @@ public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exc
         span.kind(Kind.SERVER).name(method);
         ctx.log().addListener(log -> SpanContextUtil.startSpan(span, log),
                               RequestLogAvailability.REQUEST_START);
+        ctx.log().addListener(
+                log -> SpanTags.logWireReceive(span, log.requestFirstBytesTransferredTimeNanos(), log),
+                RequestLogAvailability.REQUEST_FIRST_BYTES_TRANSFERRED);
 
         // Ensure the trace context propagates to children
         ctx.onChild(RequestContextCurrentTraceContext::copy);
 
+        ctx.log().addListener(
+                log -> SpanTags.logWireSend(span, log.responseFirstBytesTransferredTimeNanos(), log),
+                RequestLogAvailability.RESPONSE_FIRST_BYTES_TRANSFERRED);
         ctx.log().addListener(log -> SpanContextUtil.closeSpan(span, log), RequestLogAvailability.COMPLETE);
```

Comment:
How about logging wireReceive and wireSend when closing a span to reduce the number of log listeners? Is it important to add wire annotations on time?

Reply:
Good point, moved to completion listener.